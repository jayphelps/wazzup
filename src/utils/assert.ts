export const assert = (value: boolean, msg = 'Assertion failed') => {
  if (value === false) {
    throw new Error(msg);
  }
};
