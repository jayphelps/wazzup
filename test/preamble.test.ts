import { ast } from '../';
import { assertModule } from './utils/assertModule';

describe('Preamble', () => {
  let module;

  beforeEach(() => {
    module = new ast.Module();
  });

  test('contains standard preamble', () => {
    assertModule(module, `
      | (module)
    `);
  });
});
