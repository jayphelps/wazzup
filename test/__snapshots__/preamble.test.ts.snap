// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preamble contains standard preamble 1`] = `
Module {
  "exports": Array [],
  "functions": Array [],
  "globals": Array [],
  "imports": Array [],
  "memories": Array [],
  "tables": Array [],
  "types": Array [],
  "version": 1,
}
`;

exports[`Preamble contains standard preamble 2`] = `
Uint8Array [
  0,
  97,
  115,
  109,
  1,
  0,
  0,
  0,
]
`;
